### PR TITLE
[-] MO: socialsharing / change in_array() to method_exists on display…

### PR DESCRIPTION
**Pull request by PrestaEdit**
…Header

### CODE REVIEW

File: socialsharing.php [hunk 1]
```diff
@@ -152,8 +152,9 @@ public function getContent()
 
 	public function hookDisplayHeader($params)
 	{
-		if (!isset($this->context->controller->php_self) || !in_array($this->context->controller->php_self, array('product', 'products-comparison')))
+		if (!isset($this->context->controller) || !method_exists($this->context->controller, 'getProduct')) {
```
Comment (gRoussac) [inline socialsharing.php]: Hi,

This is changing behavior, does CompareController.php has a method getProduct ?

Regards

**Comment by gRoussac** [comment]
:up:

**Comment by PrestaEdit** [comment]
The controller CompareController havn't this method ; it's why - if you leave the module as before - it will do some errors on the line: $product = $this->context->controller->getProduct();

**Comment by gRoussac** [comment]
Hi,

How is the module supposed to be dispolayed in product comparison now ?

https://github.com/PrestaShop/socialsharing/pull/21/files

Regards

**Comment by gRoussac** [comment]
Hi,

Please revert his.

Regards

**Comment by gRoussac** [comment]
:up:

**Comment by PrestaEdit** [comment]
I suppose "up" is here for @jnadaud to revert and merge you own proposition, to be better :)

**Pull request merged**